lib/hooks.dart

```diff
@@ -396,3 +396,54 @@ T useContext<T>(Context<T> context) => ContextHelpers.unjsifyNewContext(React.us
 ///
 /// Learn more: <https://reactjs.org/docs/hooks-reference.html#useref>.
 Ref<T> useRef<T>([T initialValue]) => Ref.useRefInit(initialValue);
+
+/// Customizes the [ref] value that is exposed to parent components when using [forwardRef] by setting [ref.current]
+/// to the return value of [createHandle].
+///
+/// In most cases, imperative code using refs should be avoided.
+///
+/// > __Note:__ there are two [rules for using Hooks](https://reactjs.org/docs/hooks-rules.html):
+/// >
+/// > * Only call Hooks at the top level.
+/// > * Only call Hooks from inside a [DartFunctionComponent].
+///
+/// __Example__:
+///
+/// ```
+/// var FancyInput = forwardRef((props, ref) {
+///   var inputRef = useRef<InputElement>();
+///
+///   useImperativeHandle(
+///     ref,
+///     () => ({
+///       'focus': () {
+///         inputRef.current.focus();
+///       }
+///     }),
+///   );
+///
+///   return react.input({
+///     'ref': inputRef,
+///     'value': props['value'],
+///     'onChange': (e) => props['update'](e.target.value),
+///   });
+/// });
+///
+/// UseImperativeHandleTestComponent(Map props) {
+///   StateHook<String> inputValue = useState('');
+///   Ref fancyInputRef = useRef();
+///
+///   return react.Fragment({}, [
+///     FancyInput({
+///       'value': inputValue.value,
+///       'update': inputValue.set,
+///       'ref': fancyInputRef,
+///     }, []),
+///     react.button({'onClick': (_) => fancyInputRef.current['focus']()}, ['Focus Input']),
+///   ]);
+/// }
+/// ```
+///
+/// Learn more: <https://reactjs.org/docs/hooks-reference.html#useimperativehandle>.
+void useImperativeHandle(Ref ref, Function() createHandle, [List dependencies]) =>
```

Comment:
I'm not the typing master like @greglittlefield-wf is, but I feel like the `createHandle` typing could use a return type and the `dependencies` could explicitly type its `List`
```suggestion
void useImperativeHandle(Ref ref, void Function() createHandle, [List<dynamic> dependencies]) =>
```